packages/trader/src/Stores/Modules/Trading/Actions/start-date.ts

```diff
@@ -1,32 +1,25 @@
 import { ContractType } from 'Stores/Modules/Trading/Helpers/contract-type';
 import { TTradeStore } from 'Types';
 
-type TOnChangeStartDate = (
-    store: TTradeStore
-) => Promise<
-    Pick<
-        TTradeStore,
-        | 'contract_start_type'
-        | 'duration_units_list'
-        | 'duration_min_max'
-        | 'duration_unit'
-        | 'start_time'
-        | 'expiry_date'
-        | 'expiry_type'
-    > & { sessions?: TTradeStore['sessions'] }
+type TOnChangeStartDate = (store: TTradeStore) => Promise<
+    Pick<TTradeStore, 'contract_start_type' | 'duration_units_list' | 'start_time' | 'expiry_date' | 'expiry_type'> & {
+        sessions?: TTradeStore['sessions'];
+    } & {
+        duration_min_max: { [key: string]: { min: number; max: number } } | { min: number; max: number };
+    } & { duration_unit: string | number }
 >;
 type TOnChangeExpiry = (store: TTradeStore) => Promise<{
-    expiry_time: TTradeStore['expiry_time'];
-    market_open_times?: TTradeStore['market_open_times'];
-    market_close_times?: TTradeStore['market_close_times'];
+    expiry_time: TTradeStore['expiry_time'] | moment.Moment;
+    market_open_times?: TTradeStore['market_open_times'] | { open: string[]; close: string[] };
+    market_close_times?: TTradeStore['market_close_times'] | { open: string[]; close: string[] };
 }>;
 
 export const onChangeStartDate: TOnChangeStartDate = async store => {
     const { contract_type, duration_unit, start_date } = store;
-    const server_time = store.root_store.common.server_time;
+    const server_time = store.root_store?.common.server_time;
     let { start_time, expiry_type } = store;
 
-    start_time = start_time || server_time.clone().add(6, 'minute').format('HH:mm'); // when there is not a default value for start_time, it should be set more than 5 min after server_time
+    start_time = start_time || server_time?.clone().add(6, 'minute').format('HH:mm') || null; // when there is not a default value for start_time, it should be set more than 5 min after server_time
```

Comment:
@kate-deriv we don't want to add ` || null` here, let's update the types of start_time and server_time to avoid error.

Reply:
Will fix